File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    dp = [[0] * len(triangle) for _ in range(len(triangle))]
    
    # 맨 윗 숫자 넣어주기
    dp[0][0] = triangle[0][0]
    for i in range(0, len(triangle) - 1):
        for j in range(0, i+1):
            # 왼쪽
            if dp[i+1][j] < dp[i][j] + triangle[i+1][j]:
                dp[i+1][j] = dp[i][j] + triangle[i+1][j]
            # 오른쪽
            if dp[i+1][j+1] < dp[i][j] + triangle[i+1][j+1]:
                dp[i+1][j+1] = dp[i][j] + triangle[i+1][j+1]
    
    return max(dp[-1])

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function createArray(n) {
  return Array.from({ length: n }, () => Array(n).fill(0));
}

function solution(triangle) {
    let triangleLength = triangle.length
    let dp = createArray(triangleLength)
    
    // 첫 줄 초기화
    dp[0][0] = triangle[0][0]
    
    for (let i=0; i<triangleLength-1; i++) {
        for (let j=0; j<triangleLength-1; j++) {
            if (dp[i+1][j] < dp[i][j] + triangle[i+1][j]) {
                dp[i+1][j] = dp[i][j] + triangle[i+1][j]
            }
            if (dp[i+1][j+1] < dp[i+1][j+1] + triangle[i+1][j+1]){
            dp[i+1][j+1] = dp[i][j] + triangle[i+1][j+1]}
        }
    }
    
    return Math.max(...dp[triangleLength-1])
}